langchain/src/vectorstores/prisma.ts

```diff
@@ -220,14 +257,18 @@ export class PrismaVectorStore<
   }
 
   async addVectors(vectors: number[][], documents: Document<TModel>[]) {
-    const idSql = this.Prisma.raw(`"${this.idColumn}"`);
+    // table name, column name cannot be parametrised
+    // these fields are thus not escaped by Prisma and can be dangerous if user input is used
+    const idColumnRaw = this.Prisma.raw(`"${this.idColumn}"`);
+    const tableNameRaw = this.Prisma.raw(`"${this.tableName}"`);
+    const vectorColumnRaw = this.Prisma.raw(`"${this.vectorColumnName}"`);
 
     await this.db.$transaction(
       vectors.map(
         (vector, idx) => this.db.$executeRaw`
-          UPDATE ${this.tableSql}
-          SET ${this.vectorColumnSql} = ${`[${vector.join(",")}]`}::vector
-          WHERE ${idSql} = ${documents[idx].metadata[this.idColumn]}
+          UPDATE ${tableNameRaw}
+          SET ${vectorColumnRaw} = ${`[${vector.join(",")}]`}::vector
+          WHERE ${idColumnRaw} = ${documents[idx].metadata[this.idColumn]}
```

Comment:
What if `documents[idx].metadata[this.idColumn]` can't be trusted? E.g. if you're loading documents in from a different source?

Reply:
Oh, right didn't see the other comment about executeRaw only being able to execute one query. I still don't love this and it should be escaped.

Reply:
afaict `documents[idx].metadata[this.idColumn]` and `[${vector.join(",")}]` is still escaped and parametrised.